Container docs (#1539)
ICB-DCM/pyPESTO: doc/containers.rst

Containers
==========

We provide a `pyPESTO OCI image <https://hub.docker.com/r/stephanmg/pypesto>`__ through the Docker.io registry ``docker.io/stephanmg/pypesto:latest``.

A docker image build is triggered on changes/commits in the `develop` branch of pyPESTO. The container is build and pushed via a GHA to the Docker.io registry with tag ``latest``.

The image can be used by using a pull from Docker.io registry: ``docker pull docker.io/stephanmg/pypesto:latest``

The Docker image can also be built locally using the Dockerfile found here: https://github.com/ICB-DCM/pyPESTO/tree/main/docker

If you require to transfer the docker image manually, build the container using the Dockerfile, then export your docker image to `.tar` by:
`docker save -o pypesto.tar pypesto:latest`. The `.tar` file can then be transfered (e.g. with scp) to a remote location.

Run Docker image
----------------

To download / pull image: `docker pull docker.io/stephanmg/pypesto`.

> To run: `docker run --rm pypesto python3 -c 'import pypesto; print(f"PyPESTO version: {pypesto.__version__}")'`.

Reviewer: Please use double-backticks for code/verbatim.

Containers
==========

We provide a pyPESTO docker image in OCI format through the Docker.io registry `docker.io/stephanmg/pypesto:latest`.

A docker image build is triggered on changes/commits in the `develop` branch of pyPESTO. The container is build and pushed via a GHA to the Docker.io registry with tag ``latest``.

> The image can be used by using a pull from Docker.io registry: `docker pull docker.io/stephanmg/pypesto:latest`

Reviewer: ```suggestion
The image can be used by using a pull from Docker.io registry: ``docker pull docker.io/stephanmg/pypesto:latest``
```

Please single backticks.

> Docker image can also be build manually by the users using the Dockerfile found here: https://github.com/ICB-DCM/pyPESTO/tree/main/docker

Reviewer: ```suggestion
The Docker image can also be built locally using the Dockerfile found here: https://github.com/ICB-DCM/pyPESTO/tree/main/docker
```

If you require to transfer the docker image manually, build the container using the Dockerfile, then export your docker image to `.tar` by:
`docker save -o pypesto.tar pypesto:latest`. The `.tar` file can then be transfered (e.g. with scp) to a remote location.

Run Docker image
----------------

To download / pull image: `docker pull docker.io/stephanmg/pypesto`.

To run: `docker run --rm pypesto python3 -c 'import pypesto; print(f"PyPESTO version: {pypesto.__version__}")'`.

> Similar instructions for the Docker drop-in replacements Podman and Singularity/Apptainer.

Reviewer: ```suggestion
```

.. _containers:

Containers
==========

> We provide a pyPESTO docker image in OCI format through `docker.io/stephanmg/pypesto`.

Reviewer: ```suggestion
We provide a pyPESTO container image in the OCI format through `docker.io/stephanmg/pypesto` (https://hub.docker.com/r/stephanmg/pypesto).
```

> Develop versions are pushed as tag ``latest``.

Reviewer: Are they? I don't see any at https://hub.docker.com/r/stephanmg/pypesto.

Also, the workflow uses `ICB_DCM/pypesto`.

Author: Should be also published now, manually pushed latest image with commit from `develop` latest from today: commit sha 20fd9dec68ff796f5be0aa38de182be4b1d00563

> Docker image can also be build manually by the users using the Dockerfile: https://github.com/ICB-DCM/pyPESTO/tree/main/docker

Reviewer: ```suggestion
The image can also be built using the provided `Dockerfile <https://github.com/ICB-DCM/pyPESTO/blob/main/docker/Dockerfile>`_.
```

If you manually build your container and need to convert to Apptainer, first export your docker image to `.tar` by:

> `docker save -o pypesto.tar pypesto:latest` and pull the image before from docker.io by `docker pull docker.io/stephanmg/pypesto:latest`.

Reviewer: Why should one go through docker if one wants to use apptainer? `apptainer pull docker://stephanmg/pypesto:latest`? 

"first export ... pull before..." is somewhat unclear.

Author: Fixed.

Then execute for generating a `.sif` file:
``apptainer build pypesto.sif docker-archive://pypesto.tar``

Likewise for singularity:
``singularity build pypesto.sif docker-archive://pypesto.tar``

> 

Reviewer: I'd leave that out. They are interchangeable.

> 

Reviewer: Move that down under the respective heading?

Docker
------

To download / pull image: `docker pull docker.io/stephanmg/pypesto`.

To run: `docker run --rm pypesto python3 -c 'import pypesto; print(f"PyPESTO version: {pypesto.__version__}")'`.

Apptainer
---------

> Same principle as for Docker, but replace with `apptainer` commands:

Reviewer 2: Would remove that, if we still show the commands used in apptainer this is if in the best case an interesting but not useful information in a "how to" imo

Author: Okay.
